common/src/main/java/com/wynntils/core/consumers/functions/templates/Template.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.consumers.functions.templates;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Template {
+    private final List<TemplatePart> parts;
```

Comment:
Idk if you like this but you can make this an unmodifiable list and then make this class a record.